ext/TrixiConvexECOSExt.jl

```diff
@@ -0,0 +1,187 @@
+function filter_eig_vals(eig_vals, verbose, threshold = 1e-12)
+    filtered_eig_vals = Complex{Float64}[]
+
+    for eig_val in eig_vals
+        if real(eig_val) < 0 && imag(eig_val) > 0 && abs(eig_val) >= threshold
+            push!(filtered_eig_vals, eig_val)
+        end
+    end
+
+    filtered_eig_vals_count = length(eig_vals) - length(filtered_eig_vals)
+
+    if verbose
+        println("$filtered_eig_vals_count eigenvalue(s) are not passed on because " *
+                "they either are in magnitude smaller than $threshold, have positive " *
+                "real parts, or have negative imaginary parts.\n")
+    end
+
+    return length(filtered_eig_vals), filtered_eig_vals
+end
+
+# Undo normalization of stability polynomial coefficients by index factorial
+# relative to consistency order.
+function undo_normalization!(consistency_order, num_stage_evals, gamma_opt)
+    for k in (consistency_order + 1):num_stage_evals
+        gamma_opt[k - consistency_order] = gamma_opt[k - consistency_order] /
+                                           factorial(k)
+    end
+    return gamma_opt
+end
+
+# Compute stability polynomials for paired explicit Runge-Kutta up to specified consistency
+# order, including contributions from free coefficients for higher orders, and
+# return the maximum absolute value
+function stability_polynomials!(pnoms, consistency_order, num_stage_evals,
+                                normalized_powered_eigvals_scaled,
+                                gamma)
+    num_eig_vals = length(pnoms)
+
+    # Initialize with zero'th order (z^0) coefficient
+    for i in 1:num_eig_vals
+        pnoms[i] = 1.0
+    end
+
+    # First `consistency_order` terms of the exponential
+    for k in 1:consistency_order
+        for i in 1:num_eig_vals
+            pnoms[i] += normalized_powered_eigvals_scaled[i, k]
+        end
+    end
+
+    # Contribution from free coefficients
+    for k in (consistency_order + 1):num_stage_evals
+        pnoms += gamma[k - consistency_order] * normalized_powered_eigvals_scaled[:, k]
+    end
+
+    # For optimization only the maximum is relevant
+    return maximum(abs(pnoms))
+end
+
+#=
+bisection()
+
+The following structures and methods provide a simplified implementation to 
+discover optimal stability polynomial for a given set of `eig_vals`
+These are designed for the one-step (i.e., Runge-Kutta methods) integration of initial value ordinary 
+and partial differential equations.
+
+- Ketcheson and Ahmadia (2012).
+Optimal stability polynomials for numerical integration of initial value problems
+[DOI: 10.2140/camcos.2012.7.247](https://doi.org/10.2140/camcos.2012.7.247)
+=#
+
+# Perform bisection to optimize timestep for stability of the polynomial
+function bisect_stability_polynomial(consistency_order, num_eig_vals, num_stage_evals,
+                                     dtmax, dteps,
+                                     eig_vals,
+                                     verbose)
```

Comment:
I've set `verbose = false` in `methods_PERK2.jl`, specifically in the constructors, since there are some outputs from `filer_eigvals` which is called in `methods_PERK2.jl` as well. Would you like me to make further adjustments?

Reply:
All good, thanks for the explanation!

Reply:
Since this function is `export`ed I would still tend to have `verbose` a kwarg with `verbose = false` as default (or don't `export` it if it should not be part of the public API.

Reply:
Sounds reasonable to me! I agree, there is probably no need to export this function.